File: C/Searching/LinearSearch.c
#include <stdio.h>
int search(int *arr, int n, int no)
{
	int i,pos=0;
	for(i=0;i<n;i=i+1)
	{
		if(*(arr+i) == no)
		{
			pos = i;
			break;
		}
	}
	return pos;
}
int main()
{
	printf("Linear Search\n");
	printf("*************\n\n");
	int arr[100];
	int n,i,no;
	printf("Enter Size of Array : ");
	scanf("%d",&n);
	printf("\n");
	printf("Enter Numbers : \n");
	for(i=0;i<n;i=i+1)
	{
		scanf("%d",&arr[i]);
	}
	printf("\n");
	printf("Enter Number to be Searched : ");
	scanf("%d",&no);
	printf("\n");
	int pos = search(arr,n,no);
	if(pos==0)
	{
		printf("Number is not Present in Array\n");
	}
	else
	{
		printf("Number Found in Position %d\n",(pos+1));
	}
}

File: Java/Algorithms/Searching/Linear Search/LinearSearch.java
class LinearSearch
{
	public static int search(int arr[], int x)
	{
		int n = arr.length;
		for (int i = 0; i < n; i++)
		{
			if (arr[i] == x)
				return i;
		}
		return -1;
	}

	public static void main(String args[])
	{
		int arr[] = { 30, 20, 50, 10, 40 };
		int x = 10;

		int result = search(arr, x);
		if (result == -1){
			System.out.print("Element is not present in array");
    }
		else{
			System.out.print("Element is present at index "+ result);
    }
	}
}
